subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/SubsetScriptTransformer.java

```diff
@@ -90,8 +91,14 @@ public void call(SourceUnit source) throws CompilationFailedException {
         // Remove all the classes other than the main class
         source.getAST().getClasses().removeIf(classNode -> classNode != scriptClass);
 
-        // Remove all the methods from the main class
+        // Remove all the inner classes and methods from the main class
         if (scriptClass != null) {
+            Iterator<InnerClassNode> innerClasses = scriptClass.getInnerClasses();
```

Comment:
What's changed on the Groovy side that's made the inner classes appear as part of the build script now?  Were they being removed in some other way or were inner classes not considered inner classes before?

Reply:
Groovy 4 began to pull on inner classes during compilation in a way that requires they be resolved: https://github.com/apache/groovy/commit/2eb76dde8857c3f81926022e180a2541ecdbb509#diff-274fc4fc6bae462698463f05c0a81fea010871186eb70ddf5ae7f337cba3e90dR1343